### PR TITLE
feat: removing the table of content and adding the title if not present before posting to medium

### PR DESCRIPTION
**Pull request by MattiooFR**
I improved the medium plugin by using BeautifulSoup to remove the generated table of content because it's totally messing the medium article.

I also added a `<h1>` tag with the post title in the article if it's not present already because filling the title attribute in the medium api doesnt work and the article ends up being without any title.

I also formatted the previous code to be compliant with PEP style.

### CODE REVIEW

File: v8/medium/medium_plugin.py [hunk 1]
```diff
@@ -33,7 +33,9 @@
 from nikola import utils
 from nikola.plugin_categories import Command
 
-LOGGER = utils.get_logger('Medium')
+from bs4 import BeautifulSoup
```
Comment (Kwpolska) [inline v8/medium/medium_plugin.py:36]: Please use `lxml.html` instead. Nikola uses lxml for all its HTML parsing needs, and bs4 would be an extra unnecessary dependency.

**Comment by Kwpolska** [comment]
From [Medium’s API docs](https://github.com/Medium/medium-api-docs#33-posts):

> The title of the post. Note that this title is used for SEO and when rendering the post as a listing, but will not appear in the actual post—for that, the title must be specified in the content field as well. 

I knew Medium was dumb, but never that it was completely braindead.  Oh well.

File: v8/medium/medium_plugin.py (hunk 1 above)
Comment (MattiooFR) [reply v8/medium/medium_plugin.py:36]: Ok that will be my goal for tomorrow ahah I should have checked before going straight for BeautifulSoup

**Comment by MattiooFR** [comment]
My code works for the TOC made with pandoc, but I don't know how rst is generating the table of content in html so I don't know if it will works for removing the toc.

**Comment by Kwpolska** [comment]
It uses div id="contents", but I would be wary about removing it, since that ID could happen in organic content.

**Comment by MattiooFR** [comment]
Ok, just thought that I forgot to remove BeautifulSoup from the requirement.txt 🤦🏻‍♂️

**Comment by MattiooFR** [comment]
Mmh, it is not in the requirements.txt anymore, maybe I actually did remove it ^^nevermind

**Comment by MattiooFR** [comment]
Oh no even better, I actually forgot to put it there the first time ahah well at least everything is fine. Have a great day!

**Pull request merged**